test/plugin/base_test.rb

```diff
@@ -1106,6 +1106,197 @@ def test_log_entry_trace_field
+          end
+          d.run
+        end
+        verify_log_entries(n, DOCKER_CONTAINER_PARAMS)
+        assert_requested_metadata_agent_stub(IMPLICIT_LOCAL_RESOURCE_ID)
+        assert_requested_metadata_agent_stub("container.#{DOCKER_CONTAINER_ID}")
+      end
+    end
+  end
+
+  # Test jsonPayload logs from Docker container stdout / stderr.
+  def test_docker_container_stdout_stderr_logs_json_payload
+    [1, 2, 3, 5, 11, 50].each do |n|
+      new_stub_context do
+        setup_gce_metadata_stubs
+        setup_metadata_agent_stubs
+        setup_logging_stubs do
+          d = create_driver(DOCKER_CONTAINER_CONFIG)
+          n.times do
+            d.emit(docker_container_stdout_stderr_log_entry(
+                     '{"msg": "test log entry ' \
+                     "#{n}" \
+                     '", "tag2": "test", "data": ' \
+                     '5000, "severity": "WARNING"}'))
+          end
+          d.run
+        end
+        verify_log_entries(n, DOCKER_CONTAINER_PARAMS, 'jsonPayload') do |entry|
+          fields = get_fields(entry['jsonPayload'])
+          assert_equal 3, fields.size, entry
+          assert_equal "test log entry #{n}", get_string(fields['msg']), entry
+          assert_equal 'test', get_string(fields['tag2']), entry
+          assert_equal 5000, get_number(fields['data']), entry
+        end
+        assert_requested_metadata_agent_stub(IMPLICIT_LOCAL_RESOURCE_ID)
+        assert_requested_metadata_agent_stub("container.#{DOCKER_CONTAINER_ID}")
+      end
+    end
+  end
+
+  # Test logs from applications running in Docker containers. These logs have
+  # the label "logging.googleapis.com/local_resource_id" set in the format of
+  # "containerName.<container_name>".
+  def test_docker_container_application_logs
+    new_stub_context do
+      setup_gce_metadata_stubs
+      setup_metadata_agent_stubs
+      setup_logging_stubs do
+        # Metadata Agent is not enabled. Will call Docker Remote API for
+        # container info.
+        d = create_driver(ENABLE_METADATA_AGENT_CONFIG)
+        d.emit(docker_container_application_log_entry(log_entry(0)))
+        d.run
+      end
+      verify_log_entries(1, DOCKER_CONTAINER_PARAMS_WITH_NO_STREAM)
+      assert_requested_metadata_agent_stub(IMPLICIT_LOCAL_RESOURCE_ID)
+      assert_requested_metadata_agent_stub(
+        "containerName.#{DOCKER_CONTAINER_NAME}")
+    end
+  end
+
+  # Test that the 'time' field from the json record is extracted and set to
+  # entry.timestamp for Docker container logs.
+  def test_time_field_extraction_for_docker_container_logs
+    new_stub_context do
+      setup_gce_metadata_stubs
+      setup_metadata_agent_stubs
+      setup_logging_stubs do
+        d = create_driver(ENABLE_METADATA_AGENT_CONFIG)
+        d.emit(docker_container_application_log_entry(log_entry(0)))
+        d.run
+      end
+      verify_log_entries(1, DOCKER_CONTAINER_PARAMS_WITH_NO_STREAM) do |entry|
```

Comment:
Careful: since you've merged #146, this now simply verifies that there's 1 entry with a 'textPayload' field, but no longer verifies the content of that field. See, e.g., `test_timestamps`, `test_one_container_log_from_tag_stderr`, and `verify_container_logs`.

Reply:
Ah, i see. Just added the corresponding `verify_default_log_entry_text` checks.

Reply:
Oops, sorry, I should have been clearer. This check happens automatically when you don't pass a block, so you only needed to change it here.

Reply:
Ah, I see. I should have taken a look at the change in #146. Just changed it back.